libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -123,7 +125,24 @@ void SRV_Channel::calc_pwm(float output_scaled)
     if (SRV_Channel::should_e_stop(get_function()) && SRV_Channels::emergency_stop) {
         output_scaled = 0.0;
         force = true;
-    }
+        if (!estop_active) {
+             AP_Logger *logger = AP_Logger::get_singleton();
+             if (logger && logger->logging_enabled()) {
+                 logger->Write_Event(LogEvent::MOTORS_EMERGENCY_STOPPED);
+             }
+        estop_active = true;
+        }
+    } else {
+      if (SRV_Channel::should_e_stop(get_function()) && !SRV_Channels::emergency_stop) { 
+         if (estop_active) {
+             AP_Logger *logger = AP_Logger::get_singleton();
+             if (logger && logger->logging_enabled()) {
+                 logger->Write_Event(LogEvent::MOTORS_EMERGENCY_STOP_CLEARED);
+             }
+             estop_active = false;
+         }   
+      } 
+    } 
```

Comment:
This should be in the `set_emergency_stop` function.

Reply:
thanks for showing me how...